src/main/java/ch/njol/skript/Skript.java

```diff
@@ -1644,178 +1644,171 @@ public static void markErrored() {
 		Set<PluginDescriptionFile> stackPlugins = new HashSet<>();
-		for (StackTraceElement s : stackTrace) { // Look through stack trace
-			for (Entry<String,PluginDescriptionFile> e : pluginPackages.entrySet()) { // Look through plugins
-				if (s.getClassName().contains(e.getKey())) // Hey, is this plugin in that stack trace?
-					stackPlugins.add(e.getValue()); // Yes? Add it to list
+		for (StackTraceElement element : stackTrace) {
+			try {
+				Class<?> clazz = Class.forName(element.getClassName());
+				if (Plugin.class.isAssignableFrom(clazz)) {
+					Plugin plugin = JavaPlugin.getProvidingPlugin(clazz);
+					stackPlugins.add(plugin.getDescription());
+				}
+			} catch (ClassNotFoundException | NoClassDefFoundError | IllegalStateException ignored) {
 			}
 		}
+		return stackPlugins;
+	}
+
 
+	private static void logPlatformSupportInfo(String issuesUrl, String downloadUrl, Set<PluginDescriptionFile> stackPlugins) {
 		SkriptUpdater updater = Skript.getInstance().getUpdater();
 
-		// Check if server platform is supported
 		if (tainted) {
-			logEx("Skript is running with developer command-line options.");
-			logEx("If you are not a developer, consider disabling them.");
+			logEx("Skript is running with developer command-line options. Consider disabling them if not a developer.");
 		} else if (getInstance().getDescription().getVersion().contains("nightly")) {
-			logEx("You're running a (buggy) nightly version of Skript.");
-			logEx("If this is not a test server, switch to a more stable release NOW!");
-			logEx("Your players are unlikely to appreciate crashes and/or data loss due to Skript bugs.");
-			logEx("");
-			logEx("Just testing things? Good. Please report this bug, so that we can fix it before a stable release.");
-			logEx("Issue tracker: " + issuesUrl);
-		} else if (!isRunningMinecraft(1, 9)) {
-			logEx("You are running an outdated Minecraft version not supported by Skript.");
-			logEx("Please update to Minecraft 1.9.4 or later or fix this yourself and send us a pull request.");
-			logEx("Alternatively, use an older Skript version; do note that those are also unsupported by us.");
-			logEx("");
-			logEx("Again, we do not support Minecraft versions this old.");
-		} else if (!serverPlatform.supported){
-			logEx("Your server platform appears to be unsupported by Skript. It might not work reliably.");
-			logEx("You can report this at " + issuesUrl + ". However, we may be unable to fix the issue.");
-			logEx("It is recommended that you switch to Paper or Spigot, should you encounter more problems.");
+			logEx("You're running a (buggy) nightly version of Skript. If this is not a test server, switch to a stable release.");
+			logEx("Please report this bug to: " + issuesUrl);
+		} else if (!serverPlatform.supported) {
+			logEx("Your server platform appears to be unsupported by Skript. Consider switching to Paper or Spigot for better compatibility.");
```

Comment:
To avoid hardcoding `Paper or Spigot`, do you think we could just stringify a list of the *supported* server platforms?